source/blog/2024-11-05-hanami-220.html.markdown

```diff
@@ -0,0 +1,244 @@
+---
+title: "Hanami 2.2.0: Persistence pays off"
+date: 2024-11-05 12:00:00 UTC
+tags: announcements
+author: Tim Riley
+image: true
+excerpt: >
+  With our new database layer and operations, the Hanami 2 vision is complete!
+---
+
+Two years ago, we [released Hanami 2.0](/blog/2022/11/22/announcing-hanami-200/), opening a new chapter for Hanami and our vision for Ruby apps.
+
+Earlier this year, we took another step and introduced our view layer [with Hanami 2.1](/blog/2024/02/27/hanami-210/).
+
+Today we complete the vision! We are thrilled to share Hanami 2.2 with you. With this release, we introduce a powerful new database layer and a brand new tool for organizing your business logic.
+
+## Persistence pays off: Hanami’s new database layer
+
+Hanami’s new database layer gives you a clear home for all aspects of your database interactions, along with the means to build your own clean interface for your app’s business layer to consume.
+
+When you generate a new app, you’ll have a ready-to-go `DATABASE_URL` in your `.env`. Our default database for new apps is SQLite, with Postgres and MySQL also supported.
+
+You can create a migration with `bundle exec hanami generate migration`, and fill it in:
+
+```ruby
+ROM::SQL.migration do
+  change do
+    create_table :posts do
+      primary_key :id
+      column :title, :text, null: false
+    end
+  end
+end
+```
+
+Now you can migrate your database with `hanami db migrate`.
```

Comment:
```suggestion
Now you can migrate your database with `bundle exec hanami db migrate`.
```

Reply:
Thanks for all these `bundle exec` suggestions. I'm in mixed minds about this. Prefixing the `bundle exec` makes everything feel a little more ungainly, and I do think Rubyists will understand whether they need it or not.

I think we have a compromise approach already in place: `bundle exec` _is_ already in place for the very first mention of the `hanami` command, just as an indicator that it's needed. The rest can live without it, for succinctness and ease of reading.